_rules/summary-non-empty-accessible-name-2t702h.md

```diff
@@ -0,0 +1,195 @@
+
+### Bibliography
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
+- [Summary element entry for ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `summary` element has an [accessible name][] because of its text content.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 2
+
+This `summary` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<details>
+	<summary aria-label="Opening times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 3
+
+This `summary` element has an [accessible name][] because of its `aria-labelledby` attribute.
+
+```html
+<span id="opening-times">Opening times</span>
+<details>
+	<summary aria-labelledby="opening-times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 4
+
+This `summary` element has an [accessible name][] because of its text content. It does not need to be the first child element of `details`.
+
+```html
+<details>
+	<p>This is a website. We are available 24/7.</p>
+	<summary>Opening times</summary>
+</details>
+```
+
+#### Passed Example 5
+
+This first `summary` element has an [accessible name][] because of its text content. The second `summary` element is inapplicable because only the first `summary` element will be used as a control for the `details` element.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `summary` element has no [accessible name][] because it has no content or attribute that can provide it.
+
+```html
+<details>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Failed Example 2
+
+This `summary` element has an [explicit role][] of `none`. However, it is [focusable][] (by default) which causes [Presentational Roles Conflict Resolution][]. It fails because it has an empty [accessible name][].
+
+```html
+<details>
+	<summary role="none"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
```

Comment:
````suggestion

#### Failed Example 3

This first `summary` element has no [accessible name][] because it is empty. The second `summary` element is inapplicable because only the first `summary` element will be used as a control for the `details` element.

```html
<details>
	<summary></summary>
	<summary>Opening times</summary>
	<p>This is a website. We are available 24/7.</p>
</details>
```

````